packages/nodes-base/nodes/Google/Gmail/Gmail.node.ts

```diff
@@ -0,0 +1,395 @@
+							if (attachmentsUi.hasOwnProperty('attachmentsValues')
+								&& !isEmpty(attachmentsUi.attachmentsValues)) {
+								// @ts-ignore
+								attachmentsValues = attachmentsUi.attachmentsValues.map((o) => {
+									const aux: Attachments = {name: '', content: '', type: ''};
+									aux.name = o.name;
+									aux.content = o.content;
+									aux.type = o.type;
+									return aux;
+								});
+							}
+
+							if (attachmentsUi.hasOwnProperty('attachmentsBinary')
+								&& !isEmpty(attachmentsUi.attachmentsBinary)
+								&& items[i].binary) {
+								// @ts-ignore
+								attachmentsBinary = attachmentsUi.attachmentsBinary.map((o) => {
+									if (items[i].binary!.hasOwnProperty(o.property)) {
+										const aux: Attachments = {name: '', content: '', type: ''};
+										aux.name = items[i].binary![o.property].fileName || 'unknown';
+										aux.content = items[i].binary![o.property].data;
+										aux.type = items[i].binary![o.property].mimeType;
+										return aux;
+									}
+								});
+							}
+
+							qs = {
+								userId: 'me',
+								uploadType: 'media'
+							};
+
+							attachmentsList = attachmentsBinary.concat(attachmentsValues) as IDataObject[];
+						}
+					}
+
+
+					const email: Email = {
+						to: toStr,
+						cc: ccStr,
+						bcc: bccStr,
+						subject: this.getNodeParameter('subject', i) as string,
+						body: this.getNodeParameter('message', i) as string,
+						attachments: attachmentsList
+					};
+
+					//-------------------------------------------------------------
+					// Define the endpoints and the qs/body for each operation
+					//-------------------------------------------------------------
+
+					//------------------------------------
+					// Create a draft
+					//------------------------------------
+					if(operation === 'create' && resource === 'draft') {
+						// https://developers.google.com/gmail/api/v1/reference/users/drafts/create
+						endpoint = '/gmail/v1/users/me/drafts';
+						method = 'POST';
+
+						const encodedMessage = encodeEmail(email);
+
+						body = {
+							'message': {
+							'raw': encodedMessage
+							}
+						};
+
+						responseData = await googleApiRequest.call(this, method, endpoint, body, qs);
+					//------------------------------------
+					// Send a message
+					//------------------------------------
+					} else if(operation === 'create' && resource === 'message') {
+						// https://developers.google.com/gmail/api/v1/reference/users/messages/send
+						endpoint = '/gmail/v1/users/me/messages/send';
+						method = 'POST';
+
+						const encodedMessage = encodeEmail(email);
+
+						body = {
+							'raw': encodedMessage
+						};
+
+						responseData = await googleApiRequest.call(this, method, endpoint, body, qs);
+					//------------------------------------
+					// Reply to a message
+					//------------------------------------
+					} else if(operation === 'reply' && resource === 'message') {
```

Comment:
Generally yes, but for this flow, I think that processing messages and drafts together makes a lot of sense bc they are so similar. It could totally be wrong, so I will ask Ricardo.

Reply:
Ah no, I meant simply `resource === 'message' && operation === 'reply'`